pkg/vault/contracts/interfaces/IBasePool.sol

```diff
@@ -86,4 +86,7 @@ interface IBasePool is IPoolSwapStructs {
         uint256 protocolSwapFeePercentage,
         bytes memory userData
     ) external returns (uint256[] memory amountsOut, uint256[] memory dueProtocolFeeAmounts);
+
+
+    function getPoolId() external view returns (bytes32);
```

Comment:
I'm fine with adding this, but it likely requires also adding `override` in `pool-utils/BasePool` and `weighted/Weighted2Tokens`.